### ISSUE TITLE
np.reshape(array, shape) is not supported whereas array.reshape(shape) is

### ISSUE THREAD

**Issue by Rik-de-Kort** [numba/numba#2798]
MnWE:
````python
import numpy as np
from numba import njit


@njit()
def foo(arg):
    arg = np.reshape((4, 5), arg)
    return arg


if __name__=="__main__":
    bar = np.zeros(20)
    print(foo(bar))
````

Error returned:
````python
Traceback (most recent call last):
  File "C:/Users/User/Documents/numba_playground/reshape_test.py", line 13, in <module>
    print(foo(bar))
  File "C:\Users\User\Anaconda3\lib\site-packages\numba\dispatcher.py", line 330, in _compile_for_args
    raise e
  File "C:\Users\User\Anaconda3\lib\site-packages\numba\dispatcher.py", line 307, in _compile_for_args
    return self.compile(tuple(argtypes))
  File "C:\Users\User\Anaconda3\lib\site-packages\numba\dispatcher.py", line 579, in compile
    cres = self._compiler.compile(args, return_type)
  File "C:\Users\User\Anaconda3\lib\site-packages\numba\dispatcher.py", line 80, in compile
    flags=flags, locals=self.locals)
  File "C:\Users\User\Anaconda3\lib\site-packages\numba\compiler.py", line 779, in compile_extra
    return pipeline.compile_extra(func)
  File "C:\Users\User\Anaconda3\lib\site-packages\numba\compiler.py", line 362, in compile_extra
    return self._compile_bytecode()
  File "C:\Users\User\Anaconda3\lib\site-packages\numba\compiler.py", line 738, in _compile_bytecode
    return self._compile_core()
  File "C:\Users\User\Anaconda3\lib\site-packages\numba\compiler.py", line 725, in _compile_core
    res = pm.run(self.status)
  File "C:\Users\User\Anaconda3\lib\site-packages\numba\compiler.py", line 248, in run
    raise patched_exception
  File "C:\Users\User\Anaconda3\lib\site-packages\numba\compiler.py", line 240, in run
    stage()
  File "C:\Users\User\Anaconda3\lib\site-packages\numba\compiler.py", line 454, in stage_nopython_frontend
    self.locals)
  File "C:\Users\User\Anaconda3\lib\site-packages\numba\compiler.py", line 881, in type_inference_stage
    infer.propagate()
  File "C:\Users\User\Anaconda3\lib\site-packages\numba\typeinfer.py", line 846, in propagate
    raise errors[0]
  File "C:\Users\User\Anaconda3\lib\site-packages\numba\typeinfer.py", line 137, in propagate
    constraint(typeinfer)
  File "C:\Users\User\Anaconda3\lib\site-packages\numba\typeinfer.py", line 521, in __call__
    raise UntypedAttributeError(ty, self.attr, loc=self.inst.loc)
numba.errors.UntypedAttributeError: Failed at nopython (nopython frontend)
Unknown attribute 'reshape' of type Module(<module 'numpy' from 'C:\\Users\\User\\Anaconda3\\lib\\site-packages\\numpy\\__init__.py'>)
File "reshape_test.py", line 7
[1] During: typing of get attribute at C:/Users/User/Documents/numba_playground/reshape_test.py (7)
````

MWE:
````python
import numpy as np
from numba import njit


@njit()
def foo2(arg):
    arg = arg.reshape((4, 5))
    return arg


if __name__=="__main__":
    bar = np.zeros(20)
    print(foo2(bar))
````

**Comment by Rik-de-Kort** [comment]
Closed because of #2839 being merged.

**Comment by Mullahz** [comment]
I also came across a similar problem of `np.reshape` not supported by numba njit.
However, I could solve the problem by doing normalization of the data (in my case array consisting image pixel values) using the sklearn preprocessing function.
`x = blk_img[:,1]
x = preprocessing.normalize([x])
@jit(nopython=True)
def my_kernel(x):
  x1 = x.reshape(32,32)
  #x1 = np.reshape(x, (4, 5), order='F') 
  return x1
x2 = my_kernel(x)`

Here `blk_img` is a matrix of size `1024x256`. I extract the first column and reshape it to a `32x32` image block.
